File: U1TP2/C. Prototipo/Simulacion/Ejercicio 4/src/main.py
from machine import Pin, ADC, PWM
import time

# Definir el pin del potenciómetro y del LED
pot_pin = ADC(Pin(34))  # Pin analógico para el potenciómetro (GPIO 34 en ESP32)
led_pin = PWM(Pin(5))   # Pin PWM para el LED (GPIO 5 en ESP32)

while True:
    # Leer el valor del potenciómetro
    pot_value = pot_pin.read()

    # Mapear el valor del potenciómetro al rango de 0 a 1023 (PWM)
    pwm_value = int(pot_value / 4)  # La función PWM de MicroPython acepta valores de 0 a 1023

    # Establecer la intensidad del LED usando PWM
    led_pin.duty(pwm_value)

    # Pequeña pausa para evitar rebotes del potenciómetro
    time.sleep(0.1)

File: U1TP2/C. Prototipo/Simulacion/Ejercicio 4/src/main.cpp
#include <Arduino.h>

// Definir los pines del potenciómetro y del LED
const int potPin = A0;  // Pin analógico para el potenciómetro
const int ledPin = 9;   // Pin PWM para el LED

void setup() {
  // Configurar el pin del LED como salida
  pinMode(ledPin, OUTPUT);
}

void loop() {
  // Leer el valor del potenciómetro
  int potValue = analogRead(potPin);
  
  // Mapear el valor del potenciómetro al rango de 0 a 255 (PWM)
  int pwmValue = map(potValue, 0, 1023, 0, 255);
  
  // Establecer la intensidad del LED usando PWM
  analogWrite(ledPin, pwmValue);
  
  // Pequeña pausa para evitar rebotes del potenciómetro
  delay(10);
}
